pkg/network/ovn_kubernetes.go

```diff
@@ -161,17 +167,53 @@ func boostrapOVN(kubeClient client.Client) (*bootstrap.BootstrapResult, error) {
 		return nil, fmt.Errorf("unable to bootstrap OVN, no master nodes found")
 	}
 
-	ovnMasterNodes := []string{}
-	for _, masterNode := range masterNodeList.Items {
-		ovnMasterNodes = append(ovnMasterNodes, masterNode.Name)
+	ovnMasterIPs := make([]string, len(masterNodeList.Items))
+	for i, masterNode := range masterNodeList.Items {
+		var ip string
+		for _, address := range masterNode.Status.Addresses {
```

Comment:
I think this logic for Node -> IP is correct, but is there any chance that this won't match the `Pod.Status.HostIP`? That's what me match on to decide who is the raft-bootstrap node.

Reply:
It matches kubelet's current logic, but there's no guarantee kubelet's logic won't change. I'm not sure we can do better without actually deploying the pods first and seeing what `hostIP` they get though.

Reply:
This is somewhat beyond this PR, but we don't actually need to match on IP address for the pod to decide if its the master. We can match on NodeName. I guess we needed to do IP resolution because some of the OVN components don't resolve AAAA?

Reply:
We were matching on node name before. In Russell's original PR we were writing both node names and IPs into the template, and you pointed out that we could get rid of the names and do everything with IPs. But I guess then I got rid of most of the other reason for passing in the raw IPs so maybe I should switch `MASTER_IP` back to `MASTER_NODE`...

Reply:
> you pointed out that we could get rid of the names and do everything with IPs

What can I say. I'm eventually consistent...

Reply:
> maybe I should switch `MASTER_IP` back to `MASTER_NODE`

except we'd still need `MASTER_IP` as well anyway. So let's just stick with this code for now.

> I guess we needed to do IP resolution because some of the OVN components don't resolve AAAA?

OVN/OVS only does DNS resolution if you link it with libunbound. (Not sure why... maybe it wants async DNS or something?) Anyway, the RHEL packages aren't built with that, so we have to pass IP addresses. And we had problems before (in master) with trying to resolve IPs address-family-independently in the shell scripts (getting v6 when we wanted v4 etc) so we have to get the IPs from the actual node objects.

Reply:
Huh. Good to know.